command/agentproxyshared/cache/lease_cache.go

```diff
@@ -189,22 +189,39 @@ func (c *LeaseCache) PersistentStorage() *cacheboltdb.BoltStorage {
 // checkCacheForDynamicSecretRequest checks the cache for a particular request based on its
 // computed ID. It returns a non-nil *SendResponse if an entry is found.
 func (c *LeaseCache) checkCacheForDynamicSecretRequest(id string) (*SendResponse, error) {
-	return c.checkCacheForRequest(id, "")
+	return c.checkCacheForRequest(id, nil)
 }
 
 // checkCacheForStaticSecretRequest checks the cache for a particular request based on its
 // computed ID. It returns a non-nil *SendResponse if an entry is found.
-// If a token is provided, it will validate that the token is allowed to retrieve this
-// cache entry, and return nil if it isn't.
-func (c *LeaseCache) checkCacheForStaticSecretRequest(id, token string) (*SendResponse, error) {
-	return c.checkCacheForRequest(id, token)
+// If a request is provided, it will validate that the token is allowed to retrieve this
+// cache entry, and return nil if it isn't. It will also evict the cache if this is a non-GET
+// request.
+func (c *LeaseCache) checkCacheForStaticSecretRequest(id string, req *SendRequest) (*SendResponse, error) {
+	return c.checkCacheForRequest(id, req)
 }
 
 // checkCacheForRequest checks the cache for a particular request based on its
 // computed ID. It returns a non-nil *SendResponse if an entry is found.
 // If a token is provided, it will validate that the token is allowed to retrieve this
 // cache entry, and return nil if it isn't.
-func (c *LeaseCache) checkCacheForRequest(id, token string) (*SendResponse, error) {
+func (c *LeaseCache) checkCacheForRequest(id string, req *SendRequest) (*SendResponse, error) {
+	var token string
+	if req != nil {
+		token = req.Token
+		if req.Request.Method != http.MethodGet {
```

Comment:
Might be good to future-proof by also excluding MethodHead, in case KV ever supports that.

Reply:
Added! I added OPTIONS too, because why not